### PR TITLE
Add a search provider for postgresql organization site

### PR DESCRIPTION
**Pull request by justdoit0823**
It's a convenient way to search postgresql document with this.

### CODE REVIEW

**Comment by zquestz** [comment]
This PR has a number of issues.

* The import path is not correct for the new provider.
* Commits require squashing.
* Travis changes should not be included.
* .gitignore change should not be included.

**Comment by justdoit0823** [comment]
As the above said, if i can't do this pull request, how can i make the binary program works fine.Travis test on go version 1.4.3 can't pass, the go get vet command will failed.The .gitignore file changed content only excludes emacs backup file.

**Comment by zquestz** [comment]
Unfortunately due to how go handles imports, there is no way for your local test to pass and the import to be defined correctly for your new provider. You need to make sure you reference where the file will be once the PR is accepted.

File: providers/postgresql/postgresql.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+type Provider struct{}
+
+// BuildURI generates a search URL for Google.
+func (p *Provider) BuildURI(q string) string {
+	return fmt.Sprintf("http://www.postgresql.org/search/?q=%s&a=1&submit=Search", url.QueryEscape(q))
+}
+
+// Tags returns the tags relevant to this provider.
+func (p *Provider) Tags() []string {
+	return []string{"search"}
```
Comment (zquestz) [inline providers/postgresql/postgresql.go]: This shouldn't be tagged search, as this isn't a search engine. This provider probably should have an empty tag set.

File: providers/postgresql/postgresql.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+)
+
+func init() {
+	providers.AddProvider("postgresql", &Provider{})
+}
+
+// Provider merely implements the Provider interface.
+type Provider struct{}
+
+// BuildURI generates a search URL for Google.
```
Comment (zquestz) [inline providers/postgresql/postgresql.go]: Incorrect comment.

File: s.go [hunk 3]
```diff
@@ -72,6 +72,7 @@ import (
 	_ "github.com/zquestz/s/providers/phandroid"
 	_ "github.com/zquestz/s/providers/php"
 	_ "github.com/zquestz/s/providers/pinterest"
+	_ "github.com/justdoit0823/s/providers/postgresql"
```
Comment (zquestz) [inline s.go]: Fix import path to be zquestz/s

**Comment by zquestz** [comment]
I went ahead and added this provider and did the fixes. Thank you for your contribution. I am sure our users will like the postgresql documentation search. =)

**Comment by justdoit0823** [comment]
Thank you for crorecting my mistakes, i should be more careful.

**Pull request closed without merging**